### DISCUSSION TITLE
Support for Waveshare 2.13inch Touch Display

### DISCUSSION THREAD

**Discussion by Jeresmov** [jayofelony/pwnagotchi#111] [General]
Hey,
qq, will there be or is there already support for the Waveshare touch display? Unfortunately I can't get it to work :(

**Comment by jayofelony** [comment]
I have a touch, it works by using waveshare_2. Try 3 or 4 perhaps? But I have no touch functionality.

**Comment by Jeresmov** [comment]
Installed Version 2.8.9 today and changed it to Waveshar_2, 3 and 4 and none of them work. Do i need to install or change anything else?

**Reply by jayofelony** [reply]
Mine works plug n play. But nine is over 2 yrs old now. Does it say anything on the waveshare about how to test it? Do you have a link to it perhaps?

**Reply by Jeresmov** [reply]
Its this one https://www.waveshare.com/wiki/2.13inch_Touch_e-Paper_HAT_Manual
But the Version 4, it even has the same pinlayout as the non touch but i dont know why it wont work...
Ive tested it with an waveshare Demon on my Raspberry Pi4 and there everything worked fine.
I dont even need the Touch function, just bought it because of the Case

**Reply by jayofelony** [reply]
So it seems it uses a different driver than v2 display since v3. Ill look into it.

**Reply by Jeresmov** [reply]
Thank you very much :)

**Comment by jayofelony** [comment]
Well turns out it is just the same as waveshare_2 if V2 and so on for V3 and V4. So either your GPIO isnt properly soldered or display is faulty.

**Reply by Jeresmov** [reply]
I bought the normal V4 Waveshare to try it with that. It will arrive tomorrow. But ill try it later again with waveshare_2.
Probably a stupid question but should I first install the Raspberry and then customize the Config.toml file or should I create a Config.toml file myself and customize it before I insert the SD card into the Raspberry?

**Reply by jayofelony** [reply]
You can create a config.toml and save it in the bootfs folder, after first boot it will be copied to the correct folder.
